src/background/connection.js

```diff
@@ -1,14 +1,26 @@
 const CONNECTION_TIMEOUT = 20000; // 20 secs.
 
 export class ConnectionTester {
-  static run() {
+  static run(receiver) {
     log("executing a fetch to check the connection");
 
+    const proxyListener = receiver.net.newProxyRequestCallback();
```

Comment:
I know that it's silly, but we don't have any communication like that cross components. What about if we do:
`const proxyListener = receiver.syncSendEvent("proxyRequest");`

And in main.js you add a:

```
  case "proxyRequest":
    return this.net.newProxyRequestCallback();
```

Reply:
wanted to do that, yes.  I just didn't look it up if we can consume a result of an event synchronously.  will update.